DataFormats/Common/interface/ValidHandle.h

```diff
@@ -0,0 +1,66 @@
+#ifndef DataFormats_Common_ValidHandle_h
+#define DataFormats_Common_ValidHandle_h
+
+/*----------------------------------------------------------------------
+  
+Handle: Non-owning "smart pointer" for reference to products and
+their provenances.
+
+The data product is always guaranteed to be valid for this handle type.
+----------------------------------------------------------------------*/
+
+#include <utility>
+#include "DataFormats/Provenance/interface/ProductID.h"
+
+namespace edm {
+  namespace vhhelper {
+    void throwIfNotValid(const void*) noexcept(false);
+  }
+  template<typename T>
+  class ValidHandle {
+  public:
+    using element_type = T;
+    
+    ValidHandle() = delete;
+    ValidHandle(T const* prod, ProductID id) noexcept(false) :
+    product_(prod), id_(std::move(id)) {
+      vhhelper::throwIfNotValid(prod);
+    }
+    
+    //NOTE: C++ disallows references to null
+    ValidHandle(T const& prod, ProductID id) noexcept(true):
+    product_(&prod), id_(std::move(id)) {
+    }
+    ValidHandle( const ValidHandle<T>& ) = default;
+    ValidHandle<T>& operator=(ValidHandle<T> const& rhs) = default;
+    ~ValidHandle() = default;
```

Comment:
What about moves? (actually simplest way, as we just reminded ourselves, would be to remove the destructor and copy constructor/assignment definitions and leave all 5 to be generated by the compiler)

Reply:
I didn't really want moves. The moves are no different from what a copy should do and I explicitly do NOT want a move of a `ValidHandle` to cause the handle to become invalid :).

Reply:
What about explicitly defining the move operations deleted then (so that the intention is clear)?

Reply:
Ok, we discussed in length.